rest/server_context.go

```diff
@@ -929,12 +929,16 @@ func (sc *ServerContext) asyncDatabaseOnline(nonCancelCtx base.NonCancellableCon
 	if err != nil {
 		base.InfofCtx(ctx, base.KeyAll, "Error starting online processes after async initialization: %v", err)
 		atomic.CompareAndSwapUint32(&dbc.State, db.DBStarting, db.DBOffline)
+		return
+	}
+
+	if !atomic.CompareAndSwapUint32(&dbc.State, db.DBStarting, db.DBOnline) {
+		// 2nd atomic might end up being Starting here if there's a legitimate race, but it's the most we can do for CAS
+		panic(fmt.Sprintf("database state wasn't Starting during asyncDatabaseOnline Online transition... now %q", db.RunStateString[atomic.LoadUint32(&dbc.State)]))
```

Comment:
Should we `PanicfCtx` and say that we can't do anything to recover from this state in the comment?

We can't take down a database because it's already coming online, so the best we can do is take down SG?

Or do you think this is unlikely to get hit and not worth panicking about?

Reply:
Hmm... This was mostly to catch it more obviously via test if we decided to forget another `return` again. I'm not sure what we should do really. I don't expect this to end up panicing in a released version, but I might expect it in development/testing.

I agree about at least using `PanicfCtx` over `panic` though. Better chance of getting that logging captured.